src/OrchardCore/OrchardCore/Environment/Shell/Configuration/ShellConfigurationSources.cs

```diff
@@ -10,18 +10,21 @@ namespace OrchardCore.Environment.Shell.Configuration
     public class ShellConfigurationSources : IShellConfigurationSources
     {
         private readonly string _container;
+        private readonly string _environment;
 
         public ShellConfigurationSources(IHostingEnvironment hostingEnvironment, IOptions<ShellOptions> shellOptions)
         {
             // e.g., App_Data/Sites
             _container = Path.Combine(shellOptions.Value.ShellsApplicationDataPath, shellOptions.Value.ShellsContainerName);
             Directory.CreateDirectory(_container);
+            _environment = hostingEnvironment.EnvironmentName;
         }
 
         public void AddSources(string tenant, IConfigurationBuilder builder)
         {
             builder
-                .AddJsonFile(Path.Combine(_container, tenant, "appsettings.json"), optional: true);
+                .AddJsonFile(Path.Combine(_container, tenant, "appsettings.json"), optional: true)
+                .AddJsonFile(Path.Combine(_container, tenant, $"appsettings.{_environment}.json"), optional: true);
         }
 
```

Comment:
Hmm, not sure because `appsettings.json` in the tenant folder is mutable, when setup / edit a tenant this is where we save **config values** that are not pre-configured (from other immutable sources) or different. Idem for **setting values** but to the shared `tenants.json`. (I did a comment on gitter related to this).

So here, in a given env you may get a diferent value, then if you setup / edit and save the tenant this value will be saved in `appsettings.json`, then this value may be used when running in another env.

Reply:
ah, ok. I can revert that. 

If a configuration is stored in tenant appsettings, how can an environment setting override at tenant level? I will see if I can find the gitter comment and understand.

Reply:
In another config source, e.g in an `appsettings.{env}.json` at the root level or just under the `App_Data` folder. This just under the `OrchardCore` section to apply to all tenants, or under a subsection (by using the tenant name) to apply to this tenant. Note: to be recognized as a tenant subsection, you have to define a state value (used as a marker) e.g `"State": Uninitialized`.

Reply:
@jtkech Yes, I revert that then. However will wait until you have chance to review #3581 as it will break this - will need more virtual path mangling!

So **not ready**